### PR TITLE
mcc faster digital out

### PR DESCRIPTION
**Pull request by dshimaoka**
Using DaqDIn is slow in reading the current port state. Instead, use a variable to store the current state. This will enable quicker digital output, necessary to reduce frame drops. The idea was conceived by Nic Price.

### CODE REVIEW

File: +neurostim/+plugins/mcc.m [hunk 1]
```diff
@@ -126,25 +127,42 @@ function afterExperiment(o)
         function digitalOut(o,channel,value,varargin)
             % digitalOut(o,channel,value [,duration])
             % Output the value to the digital channel
-            % o.digitalOut(0,unit8(2)) will write '2' to port A
-            % o.digitalOut(3,false) will set bit #3 to false.
+            % There are two modes here:
+            % 1) if VALUE is uint8, a full byte is written to port CHANNEL
+            % 2) if VALUE is logical, the single bit is set on line CHANNEL
+            % Currently Port0/A is hard-coded as input and Port1/B as
+            % output, so CHANNEL should be 1 (port B) or 9-16 (lines 1-8 in port B)
```
Comment (bartkrekelberg) [inline +neurostim/+plugins/mcc.m:134]: That is no longer true with the DDir option?

File: +neurostim/+plugins/mcc.m [hunk 2]
```diff
@@ -126,25 +127,42 @@ function afterExperiment(o)
...
+                end
+                DaqDOut(o.daq,channel(1)>8, newValue);
+                o.dOutOld((channel(1)>8)+1) = newValue;
                 
                 if size(varargin) == 1 
                     duration = varargin{1};
                     % timer function may override other functions when time is met
                     % and could cause problems for time-critical tasks
-                    o.timer = timer('StartDelay',duration/1000,'TimerFcn',@(~,~) outputToggle(o,channel,current)); %#ok<CPROPLC>
+                    o.timer = timer('StartDelay',duration/1000,'TimerFcn',@(~,~) outputToggle(o,channel(1),newValue)); %#ok<CPROPLC>
```
Comment (bartkrekelberg) [inline +neurostim/+plugins/mcc.m:165]: why channel(1) - better to check that it is a scaler right at the start and then use that variable as is?  (same comment fo the code above where channel(1) is used.

File: +neurostim/+plugins/mcc.m (hunk 1 above)
Comment (dshimaoka) [reply +neurostim/+plugins/mcc.m:134]: That's right. The lines 133-34 will be deleted.

File: +neurostim/+plugins/mcc.m (hunk 2 above)
Comment (dshimaoka) [reply +neurostim/+plugins/mcc.m:165]: Thanks for the comment. One of motivations of the current fix is to allow multiple lines to be set at once. Thus the channels can be a vector to allow the following usage: digitalOut([line1 line2], [true false]). The lines must belong to the same port, though.

Apart from the channel dimensionality, I noticed that the 3rd input to digitalOut does not work as intended. I will fix this in another pull request.

**Pull request closed without merging**